src/offset.js

```diff
@@ -82,27 +82,28 @@ jQuery.fn.extend({
 				});
 		}
 
-		var docElem, win,
+		var docElem, win, rect,
 			elem = this[ 0 ],
-			box = { top: 0, left: 0 },
 			doc = elem && elem.ownerDocument;
 
 		if ( !doc ) {
 			return;
 		}
 
-		docElem = doc.documentElement;
+		rect = elem.getBoundingClientRect();
 
-		// Make sure it's not a disconnected DOM node
-		if ( !jQuery.contains( docElem, elem ) ) {
-			return box;
+		// Make sure element is not hidden (display: none) or disconnected
+		if ( !rect.width && !rect.height && !rect.left &&
+			!rect.top && !rect.right && !rect.bottom ) {
```

Comment:
> I don't think the extra top/left checks are worth it

Do you talk here about performance? 

There is only one possible false-positive situation when element is placed at top-left corner and screen is not scrolled (or simply position: fixed). 
If you really care about this situation we can use ```getClientRects()``` which returns 0 length list for elements which does not participate in layout, according to spec:
>1. If the element on which it was invoked does not have an associated layout box return an empty DOMRectList object and stop this algorithm.

> http://www.w3.org/TR/cssom-view/#dom-element-getclientrects

So this might be the case:
```javascript
if ( !element.getClientRects().length ) {
  // We are not interested
}
```
Also, if you do not like ```getClientRects()``` method, we can use something like ```.offsetParent``` which returns ```null``` almost in same situations (but ```getClientRects()``` is better of course):

> The **offsetParent** attribute must return the result of running these steps:
1. If any of the following holds true return null and terminate this algorithm:
* The element does not have an associated CSS layout box.
* The element is the root element.
* The element is the HTML body element.
* The element’s computed value of the position property is fixed.

> http://dev.w3.org/csswg/cssom-view/#dom-htmlelement-offsetparent

Anyway, we can be here accurate as much as we want.